### ISSUE TITLE
[llvm:x86-windows] build failure

### ISSUE THREAD

**Issue by satishbabariya** [microsoft/vcpkg#12568]
**Environment**
- OS: windows

**Failure logs** 
```log
Computing installation plan...
The following packages will be built and installed:
    llvm[clang,compiler-rt,core,disable-abi-breaking-checks,disable-assertions,disable-clang-static-analyzer,enable-rtti,lld,tools]:x86-windows
Detecting compiler hash for triplet x86-windows...
Starting package 1/1: llvm:x86-windows
Building package llvm[clang,compiler-rt,core,disable-abi-breaking-checks,disable-assertions,disable-clang-static-analyzer,enable-rtti,lld,tools]:x86-windows...
Could not locate cached archive: C:\Users\Satish\AppData\Local\vcpkg\archives\aa\aa04d482bc2b14e89d1ceae6ec27ee8a1cf12d46.zip
-- Note: llvm only supports static library linkage. Building static library.
-- Using cached C:/Users/Satish/vcpkg/downloads/llvm-llvm-project-llvmorg-10.0.0.tar.gz
-- Cleaning sources at C:/Users/Satish/vcpkg/buildtrees/llvm/src/org-10.0.0-32c002c70c.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/Users/Satish/vcpkg/downloads/llvm-llvm-project-llvmorg-10.0.0.tar.gz
-- Applying patch 0001-allow-to-use-commas.patch
-- Applying patch 0002-fix-install-paths.patch
-- Applying patch 0003-fix-vs2019-v16.6.patch
-- Using source at C:/Users/Satish/vcpkg/buildtrees/llvm/src/org-10.0.0-32c002c70c.clean
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: C:/Users/Satish/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- /p:VCPkgLocalAppDataDisabled=true /p:UseIntelMKL=No /m
    Working Directory: C:/Users/Satish/vcpkg/buildtrees/llvm/x86-windows-dbg
    See logs for more information:
      C:\Users\Satish\vcpkg\buildtrees\llvm\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/llvm/portfile.cmake:196 (vcpkg_install_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package llvm:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: llvm:x86-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.
```

**Comment by StarGate-One** [comment]
Please attach/upload `C:\Users\Satish\vcpkg\buildtrees\llvm\install-x86-windows-dbg-out.log`

**Comment by satishbabariya** [comment]
@StarGate-One  [install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4975965/install-x86-windows-dbg-out.log)

**Comment by StarGate-One** [comment]
- `fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory`
- The file missing is in the Windows SDK
- vcpkg needs to have Windows SDK version 8.1 or later installed?

**Comment by PhoebeHui** [comment]
There are 2 issues here:
1.  fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory

You need to install atl/mfc related components, for vs2017,  you can install it by vs installer:
![image](https://user-images.githubusercontent.com/20694052/88513312-a4277d80-d01a-11ea-86ce-145a7762cdf9.png)

2. error C3437: #pragma execution_character_set is not supported when /source-charset, /execution-charset, or /utf-8 has been specified,

This seem to  be an issue that need to fix with the feature 'compiler-rt', this is sencond feature added by PR #10295 that doesn't work. Another one is openmp, related issue https://github.com/microsoft/vcpkg/issues/12377

**Comment by PhoebeHui** [comment]
This issue would be fixed by PR https://github.com/microsoft/vcpkg/pull/13998.

**Comment by PhoebeHui** [comment]
@satishbabariya, the PR has been merged, please get latest source and try again.

**Comment by satishbabariya** [comment]
Thanks, @PhoebeHui I'll try again.

**Comment by knzivid** [comment]
Sorry to bump a dead thread

> error C3437: #pragma execution_character_set is not supported when /source-charset, /execution-charset, or /utf-8 has been specified

What exactly was the fix for this issue in #13998? I am building llvm with v141 toolset and I am hitting this bug. With v142, llvm builds fine. 

[This report](https://developercommunity.visualstudio.com/t/traceloggingproviderh-is-incompatible-with-utf-8/85934) says the fix went into VS2019. Does that mean using v141 to build LLVM is unsupported?
